src/python/pants/backend/python/tasks/build_local_python_distributions.py

```diff
@@ -76,9 +86,24 @@ def _create_dist(self, dist_tgt, dist_target_dir):
     # Build a whl using SetupPyRunner and return its absolute path.
     setup_runner = SetupPyRunner(dist_target_dir, 'bdist_wheel', interpreter=interpreter)
     setup_runner.run()
-    return self._get_whl_from_dir(os.path.join(dist_target_dir, 'dist'))
 
-  def _get_whl_from_dir(self, install_dir):
+  def _inject_synthetic_dist_requirements(self, dist, req_lib_addr):
+    """Inject a synthetic requirements library that references a local wheel.
+
+    :param dist: Path of the locally built wheel to reference.
+    :param req_lib_addr:  :class: `Address` to give to the synthetic target.
+    :return: a :class: `PythonRequirementLibrary` referencing the locally-built wheel.
+    """
+    base = os.path.basename(dist)
+    whl_dir = os.path.dirname(dist)
+    whl_metadata = base.split('-')
+    req_name = '=='.join([whl_metadata[0], whl_metadata[1]])
```

Comment:
As noted in other discussions today in slack and in comments on #5479, this is broken since the requirement will be stable across source file edits (until version itself is edited in setup.py). That's a pre-existing problem though.

Reply:
I assume this will be fixed by creating unique versions, rather than directly special-casing in this or other code.